pkg/secureenclave/secureenclave.m

```diff
@@ -0,0 +1,271 @@
+#import <Foundation/Foundation.h>
+#import <CoreFoundation/CoreFoundation.h>
+#import <Security/Security.h>
+#import "secureenclave.h"
+
+
+// CFStringToCString creates a copy of the bytes of data
+// converted to UTF-8 encoding.
+char* CFStringToCString(CFStringRef data) {
+  CFIndex len = CFStringGetLength(data);
+  if (!len){
+    return NULL;
+  }
+
+  CFIndex realLen = CFStringGetMaximumSizeForEncoding(len, kCFStringEncodingUTF8);
```

Comment:
I think I had already reviewed this function, but it just now occurs to me that the `+ 1` could be done here rather than adding 1 each time `realLen` is referenced.

Reply:
done